packages/client/src/lib.js

```diff
@@ -215,6 +217,50 @@ class Web3Storage {
     return res.json()
   }
 
+  /**
+   * @param {Service} service
+   * @param {{before: string, size: number}} opts
+   * @returns {AsyncIterable<Upload>}
+   */
+  static async * listIterator (service, opts) {
+    for await (const res of paginator(Web3Storage.listPage, service, opts)) {
+      if (!res.ok) {
+        throw new Error(`${res.status} ${res.statusText}`)
+      }
+      const page = await res.json()
+      for (const upload of page) {
+        yield upload
+      }
+    }
+  }
+
+  /**
+   * @param {Service} service
+   * @param {{before: string, size: number}} opts
```

Comment:
Size doesn't really make sense here as an option. We should just set it to the max size.

Reply:
...or limit the results to the size that is passed. Which might be kinda cool - you can specify any size even though the API has a size restriction.

Reply:
I think maybe i should change the name of the param to `maxResults` so not to have `size` mean either 'page size' or 'max results'